src/mp-instance.js

```diff
@@ -1333,7 +1362,12 @@ function completeSDKInitialization(apiKey, config, mpInstance) {
     // https://go.mparticle.com/work/SQDSDKS-6044
     if (!mpInstance._Store.webviewBridgeEnabled) {
         // Load any settings/identities/attributes from cookie or localStorage
-        mpInstance._Persistence.initializeStorage();
+        // FIXME: if persistence is turned off, device id never gets set
+        // FIXME: maybe we set device id in _Store and that fixes everything?
```

Comment:
Your suggestion about setting to _Store seems rigth.  Set it there and read it from there when persistence is disabled.